persistence/persistence-spark/src/main/java/org/cbioportal/persistence/spark/ClinicalDataSparkRepository.java

```diff
@@ -0,0 +1,280 @@
+    private static final String MUTATION_COUNT = "MUTATION_COUNT";
+    private static final String FRACTION_GENOME_ALTERED = "FRACTION_GENOME_ALTERED";
+
+    @Override
+    public List<ClinicalData> getAllClinicalDataOfSampleInStudy(String studyId, String sampleId, String attributeId, String projection, Integer pageSize, Integer pageNumber, String sortBy, String direction) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public BaseMeta getMetaSampleClinicalData(String studyId, String sampleId, String attributeId) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<ClinicalData> getAllClinicalDataOfPatientInStudy(String studyId, String patientId, String attributeId, String projection, Integer pageSize, Integer pageNumber, String sortBy, String direction) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public BaseMeta getMetaPatientClinicalData(String studyId, String patientId, String attributeId) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<ClinicalData> getAllClinicalDataInStudy(String studyId, String attributeId, String clinicalDataType, String projection, Integer pageSize, Integer pageNumber, String sortBy, String direction) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public BaseMeta getMetaAllClinicalData(String studyId, String attributeId, String clinicalDataType) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<ClinicalData> fetchAllClinicalDataInStudy(String studyId, List<String> ids, List<String> attributeIds, String clinicalDataType, String projection) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public BaseMeta fetchMetaClinicalDataInStudy(String studyId, List<String> ids, List<String> attributeIds, String clinicalDataType) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<ClinicalData> fetchClinicalData(List<String> studyIds, List<String> sampleIds, List<String> attributeIds,
+                                                String clinicalDataType, String projection) {
+        List<Dataset<Row>> res = new ArrayList<>();
+        for (String studyId : new HashSet<>(studyIds)) {
+            spark.sqlContext().setConf("spark.sql.caseSensitive", "false");
+            String dataFile = ParquetConstants.CLINICAL_PATIENT;
+            if (PersistenceConstants.SAMPLE_CLINICAL_DATA_TYPE.equalsIgnoreCase(clinicalDataType)) {
+                dataFile = ParquetConstants.CLINICAL_SAMPLE;
+            } 
+            Dataset<Row> clinicalData = spark.read()
+                .option("mergeSchema", true)
+                .parquet(PARQUET_DIR + ParquetConstants.STUDIES_DIR + studyId + "/" + dataFile);
+
+            if ("PATIENT".equalsIgnoreCase(clinicalDataType)) {
```

Comment:
add a comment explaining why sample clinical data is joined when queried for patient datatype